dbos/conductor.go

```diff
@@ -0,0 +1,1007 @@
+	}
+
+	return c.sendResponse(response, string(listWorkflowsMessage))
+}
+
+// handleListQueuedWorkflowsRequest handles list queued workflows requests from the conductor
+func (c *Conductor) handleListQueuedWorkflowsRequest(data []byte, requestID string) error {
+	var req listWorkflowsConductorRequest
+	if err := json.Unmarshal(data, &req); err != nil {
+		c.logger.Error("Failed to parse list queued workflows request", "error", err)
+		return fmt.Errorf("failed to parse list queued workflows request: %w", err)
+	}
+	c.logger.Debug("Handling list queued workflows request", "request", req)
+
+	// Build functional options for ListWorkflows
+	var opts []ListWorkflowsOption
+	opts = append(opts, WithLoadInput(req.Body.LoadInput))
+	opts = append(opts, WithLoadOutput(false)) // Don't load output for queued workflows
+	opts = append(opts, WithSortDesc(req.Body.SortDesc))
+
+	// Add status filter for queued workflows
+	queuedStatuses := make([]WorkflowStatusType, 0)
+	if req.Body.Status != nil {
+		// If a specific status is requested, use that status
+		status := WorkflowStatusType(*req.Body.Status)
+		if status != WorkflowStatusPending && status != WorkflowStatusEnqueued {
+			c.logger.Warn("Received unexpected filtering status for listing queued workflows", "status", status)
+		}
+		queuedStatuses = append(queuedStatuses, status)
+	}
+	if len(queuedStatuses) > 0 {
+		queuedStatuses = []WorkflowStatusType{WorkflowStatusPending, WorkflowStatusEnqueued}
+	}
+	opts = append(opts, WithStatus(queuedStatuses))
+
+	if req.Body.WorkflowName != nil {
+		opts = append(opts, WithName(*req.Body.WorkflowName))
+	}
+	if req.Body.Limit != nil {
+		opts = append(opts, WithLimit(*req.Body.Limit))
+	}
+	if req.Body.Offset != nil {
+		opts = append(opts, WithOffset(*req.Body.Offset))
+	}
+	if req.Body.StartTime != nil {
+		opts = append(opts, WithStartTime(*req.Body.StartTime))
+	}
+	if req.Body.EndTime != nil {
+		opts = append(opts, WithEndTime(*req.Body.EndTime))
+	}
+	if req.Body.QueueName != nil {
+		opts = append(opts, WithQueueName(*req.Body.QueueName))
+	}
+
+	workflows, err := c.dbosCtx.ListWorkflows(c.dbosCtx, opts...)
+	if err != nil {
+		c.logger.Error("Failed to list queued workflows", "error", err)
+		errorMsg := fmt.Sprintf("failed to list queued workflows: %v", err)
+		response := listWorkflowsConductorResponse{
+			baseResponse: baseResponse{
+				baseMessage: baseMessage{
+					Type:      listQueuedWorkflowsMessage,
+					RequestID: requestID,
+				},
+				ErrorMessage: &errorMsg,
+			},
+			Output: []listWorkflowsConductorResponseBody{},
+		}
+		return c.sendResponse(response, string(listQueuedWorkflowsMessage))
+	}
+
+	// If no queue name was specified, only include workflows that have a queue name
+	var filteredWorkflows []WorkflowStatus
+	if req.Body.QueueName == nil {
+		for _, wf := range workflows {
+			if wf.QueueName != "" {
+				filteredWorkflows = append(filteredWorkflows, wf)
+			}
+		}
+	} else {
+		filteredWorkflows = workflows
+	}
```

Comment:
Note that I'll eventually add a filter that'll go down the SQL query (while maintaining the simplified listing UX)